lib/pbench/test/unit/agent/test_tool_meister.py

```diff
@@ -15,241 +15,188 @@
             in caplog.text
         )
 
 
 class TestSendDirectory:
-    """Test create_tar in send directory of tool meister"""
+    """Test ToolMeister._send_directory()'s use of ._create_tar()"""
 
     # Record all the mock functions called by a test
-    function_called = []
+    functions_called = []
 
-    directory = tmp_dir / f"{TestCreateTar.tm_params['hostname']}"
-
-    @staticmethod
-    def add_http_mock_response(
-        uri: str, code: HTTPStatus = HTTPStatus.OK, text: str = "succeeded"
-    ):
-        responses.add(responses.PUT, uri, status=code, body=text)
+    directory = tmp_dir / f"{tm_params['hostname']}"
 
     @staticmethod
-    def fake_tar(returncode: int, stdout: bytes):
+    def fake_create_tar(returncode: int, stdout: bytes):
         def f(directory: pathlib.Path, tar_file: pathlib.Path):
             return subprocess.CompletedProcess(
                 args=[], returncode=returncode, stdout=stdout, stderr=None
             )
 
-        TestSendDirectory.function_called.append("fake_tar")
+        __class__.functions_called.append("fake_create_tar")
         return f
 
-    @staticmethod
-    def fake_unlink(*args):
-        assert args[0] == pathlib.Path(f"{TestSendDirectory.directory}.tar.xz")
-        TestSendDirectory.function_called.append("fake_unlink")
-        pass
+    @responses.activate
+    def test_tar_create_success(self, tool_meister, monkeypatch):
+        """This test should pass the tar creation in send directory"""
 
-    @staticmethod
-    def fake_open(*args):
-        assert args[0] == pathlib.Path(f"{TestSendDirectory.directory}.tar.xz")
-        TestSendDirectory.function_called.append("fake_open")
-        return io.StringIO()
+        def fake_unlink(*args):
+            assert args[0] == pathlib.Path(f"{self.directory}.tar.xz")
+            self.functions_called.append("fake_unlink")
 
-    def fake_rmtree(self, directory: pathlib.Path):
-        assert directory == tmp_dir
-        self.function_called.append("fake_rmtree")
-        pass
+        def fake_open(*args):
+            assert args[0] == pathlib.Path(f"{self.directory}.tar.xz")
+            self.functions_called.append("fake_open")
+            return io.StringIO()
 
-    def fake_md5(self, tar_file: pathlib.Path):
-        assert tar_file == pathlib.Path(f"{self.directory}.tar.xz")
-        self.function_called.append("fake_md5")
-        return 10, "random_md5"
+        def fake_rmtree(directory: pathlib.Path):
+            assert directory == tmp_dir
+            self.functions_called.append("fake_rmtree")
 
-    @responses.activate
-    def test_tar_create_success(self, monkeypatch, logger_fixture):
-        """This test should pass the tar creation in send directory"""
+        def fake_md5(tar_file: pathlib.Path):
+            assert tar_file == pathlib.Path(f"{self.directory}.tar.xz")
+            self.functions_called.append("fake_md5")
+            return 10, "random_md5"
```

Comment:
Right, making this local to this test function make sense, no other tests are using this mocks and it keeps the unnecessary complexity out.